cachix/src/Cachix/Client/Commands.hs

```diff
@@ -235,15 +243,13 @@ pushStorePath env name storePath = do
         let newClientEnv = (clientenv env) {
               baseUrl = (baseUrl (clientenv env)) { baseUrlHost = toS name <> "." <> baseUrlHost (baseUrl (clientenv env))}
             }
-        -- TODO: http retry: retry package?
         NoContent <- escalate <=< (`runClientM` newClientEnv) $ Api.createNar
           (cachixBCClient name)
           (contentType (Proxy :: Proxy Api.XNixNar), conduitToStreaming)
         closeStdout
         exitcode <- waitForStreamingProcess cph
         -- TODO: print process stderr?
-        when (exitcode /= ExitSuccess) $
-          panic $ "Failed with " <> show exitcode <> ": " <> show cmd
+        when (exitcode /= ExitSuccess) $ throwM exitcode
```

Comment:
This is like calling `exitWith`. Wrap this exception in something that is more informative, including which command was run.